solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/SolrFeature.java

```diff
@@ -259,84 +257,48 @@ public float score() throws IOException {
               + name, e);
         }
       }
+      
+    }
 
-      @Override
-      public DocIdSetIterator iterator() {
-        return itr;
-      }
-
-      @Override
-      public int docID() {
-        return itr.docID();
-      }
-
-      private class SolrFeatureScorerIterator extends DocIdSetIterator {
-
-        final private DocIdSetIterator filterIterator;
-        final private DocIdSetIterator scorerFilter;
-        int docID;
-
-        SolrFeatureScorerIterator(DocIdSetIterator filterIterator,
-            DocIdSetIterator scorerFilter) {
-          this.filterIterator = filterIterator;
-          this.scorerFilter = scorerFilter;
-        }
-
-        @Override
-        public int docID() {
-          return filterIterator.docID();
-        }
-
-        @Override
-        public int nextDoc() throws IOException {
-          docID = filterIterator.nextDoc();
-          scorerFilter.advance(docID);
-          return docID;
-        }
-
-        @Override
-        public int advance(int target) throws IOException {
-          // We use iterator to catch the scorer up since
-          // that checks if the target id is in the query + all the filters
-          docID = filterIterator.advance(target);
-          scorerFilter.advance(docID);
-          return docID;
-        }
+    private class SolrFeatureScorerIterator extends DocIdSetIterator {
 
-        @Override
-        public long cost() {
-          return 0; // FIXME: Make this work?
-        }
+      final private DocIdSetIterator filterIterator;
+      final private DocIdSetIterator scorerFilter;
+      int docID;
```

Comment:
Do we need  `docID` to be an instance variable? it seems used only to copy values in the methods.. I think you can remove it and declare it in the methods when you need (and probably you can use final)